src/cache-utils.ts

```diff
@@ -110,6 +110,20 @@ export const getPackageManagerInfo = async (packageManager: string) => {
   }
 };
 
+/**
+ * getProjectDirectoriesFromCacheDependencyPath is called twice during `restoreCache`
+ *  - first through `getCacheDirectories`
+ *  - second from `repoHasYarn3ManagedCache`
+ *
+ *  it contains expensive IO operation and thus should be memoized
```

Comment:
Is this a typo?
```suggestion
 *  it contains expensive IO operation and thus should be memorised
```

Reply:
No, it is by called by intention https://en.wikipedia.org/wiki/Memoization

Reply:
@MaksimZhukov nope https://en.wikipedia.org/wiki/Memoization